CMakeLists.txt

```diff
@@ -870,22 +871,23 @@ if (ENABLE_FUZZING)
   add_subdirectory(fuzz)
 endif ()
 
+# add headers sanity check target that includes all headers independently	
+set(check_headers_dir "${PROJECT_BINARY_DIR}/check-headers")	
+file(GLOB_RECURSE headers_to_check	
+  ${PROJECT_BINARY_DIR}/*.hpp	
+  ${PROJECT_SOURCE_DIR}/include/*.hpp)	
+foreach(header ${headers_to_check})	
+  if ("${header}" MATCHES ".*/include/nodejs/.*")
+    # we do not check NodeJS bindings headers
+    continue()
+  endif()
+  get_filename_component(filename ${header} NAME_WE)	
+  set(filename "${check_headers_dir}/${filename}.cpp")	
+  if (NOT EXISTS ${filename})	
+    file(WRITE ${filename} "#include \"${header}\"\n")	
+  endif()	
+  list(APPEND sources ${filename})	
+endforeach()	
+add_library(check-headers STATIC EXCLUDE_FROM_ALL ${sources})	
+set_target_properties(check-headers PROPERTIES ARCHIVE_OUTPUT_DIRECTORY ${check_headers_dir})	
```

Comment:
It is something I had to spend a lot of time on. 

First of all as I understand the only purpose of this is to check "sanity" of headers, i.e. that every header includes everything what is needed to compile it. That's why it is not clear why we had condition to only have it if `ENABLE_NODE_BINDINGS=ON`. The second as it turned out we never built it due to `EXCLUDE_FROM_ALL`, but MSVC builds it for some reason and that's why I faced it and had to modify some headers in this PR.